### PR TITLE
Added an option to "Add yourself" as author during abstract submission

### PR DESCRIPTION
**Pull request by jgrigera**
Thanks for considering this.

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Could you please include a screenshot how it looks like in the UI? Also, the indentation of the code you added is a bit strange/inconsistent.

**Comment by jgrigera** [comment]
Sure, here is the Screenshot:

Screenshot_2020-04-16 Seventeenth Annual Conference

Also, apologies for the indentation mess - flake sorts out all my python but emacs can be sometimes idiosyncratic with templates and javascript.

File: indico/web/templates/forms/_person_link_widget_base.html [hunk 1]
```diff
@@ -68,6 +72,9 @@
                 secondaryAuthor: {{ field.require_secondary_author | default(false) | tojson }},
                 submitter: {{ field.require_submitter_author | default(false) | tojson }},
                 speaker: {{ field.require_speaker_author | default(false) | tojson }}
+            },
+            sessionUser: {
+                email: {{ session.user.email | default(false) | tojson }}
```
Comment (ThiefMaster) [inline indico/web/templates/forms/_person_link_widget_base.html:77]: No need for the default - if you see this field you are always logged in, so the value won't ever be undefined.

File: indico/web/client/js/jquery/widgets/jinja/person_link_widget.js [hunk 2]
```diff
@@ -381,6 +382,20 @@
...
+    $buttonAddMyself.on('click', function() {
+      indicoRequest('search.users', options.sessionUser, function(result, error) {
+          if (!error) {
+            if (result.length == 0) {
+              self.appendMessage("I cant seem to find you...");
+            } else {
+              $field.principalfield('add', result);
+            }
+          } else {
+              self.appendMessage("Error searching for you");
```
Comment (ThiefMaster) [inline indico/web/client/js/jquery/widgets/jinja/person_link_widget.js:394]: Too much indentation.

File: indico/web/client/js/jquery/widgets/jinja/person_link_widget.js [hunk 3]
```diff
@@ -381,6 +382,20 @@
       $field.principalfield('enter');
     });
 
+    $buttonAddMyself.on('click', function() {
+      indicoRequest('search.users', options.sessionUser, function(result, error) {
```
Comment (ThiefMaster) [inline indico/web/client/js/jquery/widgets/jinja/person_link_widget.js:386]: Why not pass all the data you get in the result via `options.sessionUser` and thus completely avoid the legacy API request?

File: indico/web/templates/forms/_person_link_widget_base.html [hunk 4]
```diff
@@ -29,7 +29,11 @@
                 {% else %}
                     {% set msg = _("Search among Indico users") %}
                 {% endif %}
-                <span id="add-existing-{{ field.id }}" class="i-button highlight text-color {{ 'disabled' if field.disable_user_search }}"
+                <span id="add-myself-{{ field.id }}" class="i-button highlight text-color"
+                       title="{% trans %}Add yourself as author{% endtrans %}">
```
Comment (ThiefMaster) [inline indico/web/templates/forms/_person_link_widget_base.html:33]: 1 space too much indentation

File: indico/web/templates/forms/_person_link_widget_base.html [hunk 5]
```diff
@@ -29,7 +29,11 @@
...
+                     {% trans %}Add Myself{% endtrans %}
```
Comment (ThiefMaster) [inline indico/web/templates/forms/_person_link_widget_base.html:34]: *Add myself*

File: indico/web/client/js/jquery/widgets/jinja/person_link_widget.js [hunk 6]
```diff
@@ -381,6 +382,20 @@
...
+          if (!error) {
+            if (result.length == 0) {
```
Comment (mic4ael) [inline indico/web/client/js/jquery/widgets/jinja/person_link_widget.js:388]:
```suggestion
            if (result.length === 0) {
```

File: indico/web/client/js/jquery/widgets/jinja/person_link_widget.js [hunk 7]
```diff
@@ -381,6 +382,20 @@
...
+              self.appendMessage("I cant seem to find you...");
```
Comment (mic4ael) [inline indico/web/client/js/jquery/widgets/jinja/person_link_widget.js:389]:
```suggestion
              self.appendMessage("I can't seem to find you...");
```

File: indico/web/client/js/jquery/widgets/jinja/person_link_widget.js (hunk 2 above)
Comment (mic4ael) [inline indico/web/client/js/jquery/widgets/jinja/person_link_widget.js:394]:
```suggestion
              self.appendMessage('Error searching for you');
```

**Comment by ThiefMaster** [comment]
Hey @jgrigera, just wanted to check whether you are going to update the PR based on the feedback at some point :)

File: indico/web/templates/forms/_person_link_widget_base.html (hunk 1 above)
Comment (jgrigera) [reply indico/web/templates/forms/_person_link_widget_base.html:77]: Great - will do this

File: indico/web/client/js/jquery/widgets/jinja/person_link_widget.js (hunk 3 above)
Comment (jgrigera) [reply indico/web/client/js/jquery/widgets/jinja/person_link_widget.js:386]: Well, I'd say that the whole person_link_widget uses this legacy API so I just followed what was done in search. I understand that could be a better approach, but was not sure of how to build the right data for add so resorted to this call that seemed consistent with the rest. I do not feel I could do what you suggest, despite it seems reasonable, sorry!

**Comment by jgrigera** [comment]
Sorry for the delay - just did with the exception of changing the legacy API bit, thanks.

**Comment by ThiefMaster** [comment]
Thanks for the PR. I just fixed a few things (and removed the legacy API call) and merged it, so this will be included in v2.3.

**Pull request merged**